pkg/emulator/libretro/image/scale.go

```diff
@@ -0,0 +1,80 @@
+package image
+
+import (
+	"image"
+)
+
+const (
+	// skips image interpolation
+	ScaleSkip = -1
+	// initial image interpolation algorithm
+	ScaleOld = 0
+	// nearest neighbour interpolation
+	ScaleNearestNeighbour = 1
+)
+
+func Resize(scaleType int, fn Format, w int, h int, packedW int, vw int, vh int, bpp int, data []byte, image *image.RGBA) {
+	// !to do set it once instead switching on each iteration
+	// !to do skip resize if w=vw h=vh
+	switch scaleType {
+	case ScaleSkip:
```

Comment:
If you want to make it looks Go. You can have an interface with resize signature. 
type resizer interface {
  func resize(fn, w, h, packedW, vw, vh, bpp, data, image)
}